src/components/CodeBox.vue

```diff
@@ -98,60 +95,28 @@ watch(
   <div class="codebox">
     <div class="code-line" v-for="line in codeLines" :key="line">
       <div class="btn-group" v-for="word in line" :key="word">
-        <div v-if="word.word == tab" class="tab"></div>
-        <button
-          :id="'word-' + word.id"
-          v-if="word.word != tab && word.word != newLine"
-          @click="clickedButton(word)"
-          class="code-word"
-          :class="{
-            'code-space': word.word == space && !isWordSelectedBug(word.id),
-            'code-space-selected': word.word == space && isWordSelectedBug(word.id),
-            'right-code': codeFeedback.hasFeedback(word.id) && codeFeedback.getFeedback(word.id).success,
-            'wrong-code': codeFeedback.hasFeedback(word.id) && !codeFeedback.getFeedback(word.id).success,
-            'selected-code': !submitted && isWordSelectedBug(word.id),
-          }"
-        >
-          <b-badge v-if="word.word == space && !isWordSelectedBug(word.id)" variant="success" class="code-space-badge">+</b-badge>
-          <pre
-            v-else
-            v-highlightjs
-          ><code v-if="!isWordSelectedBug(word.id)" class="java">{{ word.word }}</code><code v-else>{{ getCorrectedWordValue(word.id) }}</code></pre>
-        </button>
-        <b-popover
-          v-if="codeFeedback.hasFeedback(word.id)"
-          :target="'word-' + word.id"
-          triggers="hover"
-          placement="top"
-          variant="danger"
-        >
-          <template #title>Code Feedback</template>
-          Selected right: <a v-if="codeFeedback.getFeedback(word.id).codeSelectedSuccessful" class="text-success">Successful</a>
-          <a v-else class="text-danger">Failed</a><br />
-          Error Type: <a v-if="codeFeedback.getFeedback(word.id).codeErrorTypeSuccessful" class="text-success">Successful</a>
-          <a v-else class="text-danger">Failed</a><br />
-          Code Fixed: <a v-if="codeFeedback.getFeedback(word.id).codeFixedSuccessful" class="text-success">Successful</a>
-          <a v-else class="text-danger">Failed</a><br />
-        </b-popover>
+        <WordBox
+          :word="word"
+          :codeFeedback="codeFeedback"
+          :selectBugWord="selectBugWord"
+          :selected="isSelectedBugWord(word.id)"
+          :correctedBugWordValue="getCorrectedBugWordValue(word.id)"
```

Comment:
Is this the `correct` answer (given by the lecturer) or the `corrected` answer (by the user)?
These two letters make a difference in this case...

Reply:
the correct answer by the user. The feedback (which includes failed or succeed) is stored in `codeFeedback`, but we do not store there the exact solution, just if it was right or wrong

Reply:
Is this solved?
They make a difference and are used the correct way like @maexled  mentioned